### PR TITLE
Switch default time format for ingest from Joda to Java for v7

### PR DESCRIPTION
**Pull request by droberts195**
Date formats with and without the "8" prefix are now all treated
as Java time formats, so that ingest does the same as mappings
in this respect.

Relates #27330

### CODE REVIEW

**Comment by droberts195** [comment]
I opened issue #37935 to discuss side effects of the fact that this change also changes the formats used for date index naming.

File: modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateFormat.java [hunk 1]
```diff
@@ -73,30 +72,22 @@ private long parseMillis(String date) {
...
+
+            int year = LocalDate.now(ZoneOffset.UTC).getYear();
+            DateFormatter formatter = DateFormatter.forPattern(format)
+                .withLocale(locale)
+                .withZone(DateUtils.dateTimeZoneToZoneId(timezone));
+            return text -> {
+                ZonedDateTime defaultZonedDateTime = Instant.EPOCH.atZone(ZoneOffset.UTC).withYear(year);
+                TemporalAccessor accessor = formatter.parse(text);
+                long millis = DateFormatters.toZonedDateTime(accessor, defaultZonedDateTime).toInstant().toEpochMilli();
+                return new DateTime(millis, timezone);
```
Comment (spinscale) [inline modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateFormat.java]: we are still returned a joda time class here, that we need to get rid of. I am happy to fix this in a follow up PR though

**Review by spinscale — APPROVED** [review]
I left one comment, as we need to do some more work here, but I can just do that as well.

File: modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateFormat.java (hunk 1 above)
Comment (droberts195) [reply modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateFormat.java]: Thanks @spinscale.  I'd prefer to leave that for a followup PR as I'm sure removing Joda classes will spread out far and wide through the codebase.

**Comment by droberts195** [comment]
I'll hold off from merging this until somebody has commented on #37935.

**Pull request merged**